pkg/controller/instance/controller_reconcile.go

```diff
@@ -369,6 +381,18 @@ func (igr *instanceGraphReconciler) deleteResourcesInOrder(ctx context.Context)
 			continue
 		}
 
+		// Skip deletion for read-only resources
+		if igr.runtime.ResourceDescriptor(resourceID).IsExternalRef() {
+			igr.state.ResourceStates[resourceID].State = "SKIPPED"
```

Comment:
shall we use enums since this patch is merged https://github.com/kro-run/kro/blob/main/pkg/controller/instance/controller_reconcile.go#L35-L47

Reply:
missed it during merge.